drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -457,11 +500,12 @@ def _is_finished(self):
         completed_count = 0
         total_count = 0
         item_state = self._get_item_state()
+
         for item in self.data['items']:
-            if item['zone'] != 'none':
+            item_id = item['id']
+            if self._get_valid_item_zones(item_id) != []:
                 total_count += 1
-                item_id = str(item['id'])
-                if item_id in item_state:
+                if str(item_id) in item_state:
                     completed_count += 1
```

Comment:
So, I definitely could make that change, and it's something I thought about, but it kind of seems like that might be a good abstraction to design as part of adding Assessment Mode, because it seems likely to turn `_get_grade` and `_is_finished` on their respective heads anyway.